### ISSUE TITLE
encounter a error when I run the baselines

### ISSUE THREAD

**Issue by sunxin000** [qitianwu/GraphOOD-GNNSafe#2]
<img width="1029" alt="image" src="https://github.com/qitianwu/GraphOOD-GNNSafe/assets/48592085/64763829-5e6e-4db6-95e6-01f8e212b59d">
baselines.py does not import autograd and Variable, so I import them from pytorch. However, after I import them, there still occur an error.

**Comment by Ytchen981** [comment]
This error occurs because the function evaluate_detect in data_utils.py is "@torch.no_grad()"  while the baseline model ODIN and Mahalanobis caculates gradient during detection. I have fixed the error and update the code. The simplest way to fix this is remove the "@torch.no_grad()" and do not use torch.no_grad() for ODIN and Mahalanobis. Thanks for letting us know.
